packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-block-convert-button.js

```diff
@@ -98,15 +107,61 @@ export default function ReusableBlockConvertButton( {
 	return (
 		<BlockSettingsMenuControls>
 			{ ( { onClose } ) => (
-				<MenuItem
-					icon={ reusableBlock }
-					onClick={ () => {
-						onConvert();
-						onClose();
-					} }
-				>
-					{ __( 'Add to Reusable blocks' ) }
-				</MenuItem>
+				<>
+					<MenuItem
+						icon={ reusableBlock }
+						onClick={ () => {
+							setIsModalOpen( true );
+						} }
+					>
+						{ __( 'Add to Reusable blocks' ) }
+					</MenuItem>
+					{ isModalOpen && (
+						<Modal
+							title={ __( 'Create reusable block' ) }
+							closeLabel={ __( 'Close' ) }
+							onRequestClose={ () => {
+								setIsModalOpen( false );
+								setTitle( '' );
+							} }
+							overlayClassName="reusable-blocks-menu-items__convert-modal"
+						>
+							<form
+								onSubmit={ ( event ) => {
+									event.preventDefault();
+									onConvert( title );
+									setIsModalOpen( false );
+									setTitle( '' );
+									onClose();
+								} }
+							>
+								<TextControl
+									label={ __( 'Name your reusable block' ) }
```

Comment:
This could be just "Add name" or "Name"

Reply:
Sure, I just used what was shown on the issue's video :)